commit ee2a37ab39b013589647ced2e6526c7358cb2111
Author: Takashi Iwai <tiwai@suse.de>
Date:   Thu Mar 10 12:02:49 2016 +0100

    ALSA: hda - Don't handle ELD notify from invalid port
    
    commit 4f8e4f3537cafc4de128e6bfdf83baa78bc60eb1 upstream.
    
    The current Intel HDMI codec driver supports only three fixed ports
    from port B to port D.  However, i915 driver may assign a DP on other
    ports, e.g. port A, when no eDP is used.  This incompatibility is
    caught later at pin_nid_to_pin_index() and results in a warning
    message like "HDMI: pin nid 4 not registered" at each time.
    
    This patch filters out such invalid events beforehand, so that the
    kernel won't be too grumbling.
    
    Reported-by: Stefan Assmann <sassmann@kpanic.de>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 70c9456..f7bcd8d 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -2353,6 +2353,10 @@ static void intel_pin_eld_notify(void *audio_ptr, int port)
 	struct hda_codec *codec = audio_ptr;
 	int pin_nid = port + 0x04;
 
+	/* we assume only from port-B to port-D */
+	if (port < 1 || port > 3)
+		return;
+
 	/* skip notification during system suspend (but not in runtime PM);
 	 * the state will be updated at resume
 	 */
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1353_ee2a37ab39b013589647ced2e6526c7358cb2111.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1353_ee2a37ab39b013589647ced2e6526c7358cb2111.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
